signer/src/testing/dummy.rs

```diff
@@ -362,14 +365,24 @@ impl fake::Dummy<SignerSetConfig> for RotateKeysTransaction {
         let signer_set: Vec<PublicKey> = std::iter::repeat_with(|| fake::Faker.fake_with_rng(rng))
             .take(config.num_keys as usize)
             .collect();
-        let aggregate_key = PublicKey::combine_keys(signer_set.iter()).unwrap();
+
         let address = StacksPrincipal::from(clarity::vm::types::PrincipalData::from(
-            StacksAddress::p2pkh(false, &aggregate_key.into()),
+            StacksAddress::from_public_keys(
+                C32_ADDRESS_VERSION_TESTNET_MULTISIG,
+                &AddressHashMode::SerializeP2SH,
+                config.signatures_required as usize,
+                &signer_set
+                    .iter()
+                    .map(Secp256k1PublicKey::from)
+                    .collect::<Vec<_>>(),
+            )
+            .expect("failed to create StacksAddress"),
         ));
+
         RotateKeysTransaction {
             txid: fake::Faker.fake_with_rng(rng),
             address,
-            aggregate_key,
+            aggregate_key: PublicKey::combine_keys(signer_set.iter()).unwrap(),
```

Comment:
Although this is what it was before, you should actually make this a random public key, since that is what it basically is. It is never the combined keys of the signer set public keys.

Reply:
+1 great point!